### PR TITLE
Add Tensorflow GraphDef saving into asdf

### PR DESCRIPTION
**Pull request by warenlg**

### CODE REVIEW

File: sourced/ml/models/graph_def.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from modelforge import Model
+
+
+@register_model
+class GraphDef(Model):
+    """
+    Id split model exported a a GraphDef protobuf.
```
Comment (vmarkovtsev) [inline sourced/ml/models/graph_def.py]: The name is too generic for such a narrow model. I do not see the Tensorflow serialization here, it should be inside. Also deserialization.

So I would split it into the generic model - please do not name it GraphDef, it collides with the real class name from Tensorflow, - and the concrete model. The latter is out of this particular PRs scope so please do not add it here.

The idsplit model should be self contained. In other words, I expect it to contain the "split" method, or even `__call__`

Comment (EgorBu) [reply sourced/ml/models/graph_def.py]: I would recommend making two class: one generic to save/load tensorflow graphs, second for working with keras model as @vmarkovtsev suggested.
Second class should mimic functions from keras - just forward all arguments - it should be easy and convenient to use.
And you could create third class specific for id-splitter only to be optimized for our use cases (convert string to proper input and so on, convert predictions to proper output).

Comment (vmarkovtsev) [reply sourced/ml/models/graph_def.py]: Egor, there is no Keras here. It is the lower level GraphDef representation.

Comment (vmarkovtsev) [reply sourced/ml/models/graph_def.py]: We are using pure Tensorflow here.

**Comment by warenlg** [comment]
Just to know if I'm in the right direction, is it where you want the serialization ?

File: sourced/ml/models/tensorflow_model.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+
+@register_model
+class TensorFlowModel(Model):
+    """
+    TensorFlow Protobuf model exported in the ASDS format with GraphDef inside.
+    """
+    NAME = "tensorflow-model"
+
+    def construct(self, tensorflow_model=None, session=None, list_outputs=None):
```
Comment (vmarkovtsev) [inline sourced/ml/models/tensorflow_model.py]: Remove `tensorflow_model`. You can take it from the session.
`session: "tensorflow.Session"`
`list_outputs` > `outputs: List[str]`

Mind the strings - we do not want to rigidly depend on Tensorflow.

File: sourced/ml/models/tensorflow_model.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        self._tensorflow_model = tensorflow_model
+        self._session = session
+        self._list_outputs = list_outputs
+        return self
+
+    @property
+    def tensorflow_model(self):
+        """
+        Returns a tensorflow model either serialized or deserialized.
```
Comment (vmarkovtsev) [inline sourced/ml/models/tensorflow_model.py]: There should be no "either". Always deserialized.

File: sourced/ml/models/tensorflow_model.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        """
+        return self._tensorflow_model
+
+    @property
+    def session(self):
```
Comment (vmarkovtsev) [inline sourced/ml/models/tensorflow_model.py]: This is not a property of this model

File: sourced/ml/models/tensorflow_model.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
+from modelforge import register_model, Model
+
+
+@register_model
+class TensorFlowModel(Model):
+    """
+    TensorFlow Protobuf model exported in the ASDS format with GraphDef inside.
```
Comment (vmarkovtsev) [inline sourced/ml/models/tensorflow_model.py]: ASDS > Modelforge

File: sourced/ml/models/tensorflow_model.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    """
+    NAME = "tensorflow-model"
+
+    def construct(self, tensorflow_model=None, session=None, list_outputs=None):
+        self._tensorflow_model = tensorflow_model
```
Comment (vmarkovtsev) [inline sourced/ml/models/tensorflow_model.py]: `_tensorflow_model` > `_graphdef`

**Comment by vmarkovtsev** [comment]
@warenlg I implemented this myself.

**Pull request closed without merging**